src/plaid/plaid.service.ts

```diff
@@ -0,0 +1,79 @@
+import { Injectable } from '@nestjs/common'
+import { ModelId } from '../core/model'
+import { EncryptionService } from '../encryption/encryption.service'
+import { PlaidItem, tableName } from './plaidItem.entity'
+import { InjectRepository } from '@nestjs/typeorm'
+import { Repository } from 'typeorm'
+import plaid, { Account, Client } from 'plaid'
+import { ConfigService } from '@nestjs/config'
+import { IPlaidConfig } from '../config/plaidConfig'
+
+@Injectable()
+export class PlaidService {
+  private readonly client: Client
+  constructor(
+    private readonly encryptionService: EncryptionService,
+    private readonly configService: ConfigService,
+    @InjectRepository(PlaidItem) private readonly plaidSessionRepo: Repository<PlaidItem>,
+  ) {
+    const config = this.configService.get<IPlaidConfig>('plaid')
+    if (!config) {
+      throw new Error('Missing Plaid Config')
+    }
+    this.client = new plaid.Client(
+      config.clientId,
+      config.secret,
+      config.publicKey,
+      plaid.environments[config.env],
+      { version: '2019-05-29', clientApp: 'Bitmatica' },
+    )
+  }
+
+  async exchangePublicToken(userId: ModelId, publicToken: string): Promise<string> {
+    const response = await this.client.exchangePublicToken(publicToken)
+    const item = new PlaidItem()
+    item.userId = userId
+    item.itemId = response.item_id
+    item.accessToken = await this.encryptionService.encrypt(
+      response.access_token,
+      PlaidService.encryptionContext(userId, response.item_id),
+    )
+    await this.plaidSessionRepo.save(item)
+    return item.itemId
+  }
+
+  async getPlaidAccounts(userId: ModelId, itemId: string): Promise<Array<Account>> {
+    const accessToken = await this.getAccessToken(userId, itemId)
+    if (!accessToken) {
+      return Promise.reject('Access token invalid')
```

Comment:
this seems weird to me - i think you should be able to throw a normal error instead of promise.rejecting
(this is a question :) )

Reply:
Updated to use `getOrThrow` here as well.